rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -207,6 +212,46 @@ public void receiveHeaders(BtcBlock[] headers) throws IOException, BlockStoreExc
         }
     }
 
+    /**
+     * Receives only one header of serialized Bitcoin block headers and adds them to the internal BlockChain structure.
+     * @param header The bitcoin headers
+     */
+    public Integer receiveHeader(BtcBlock header) throws IOException, BlockStoreException {
+        Context.propagate(btcContext);
+        this.ensureBtcBlockChain();
+
+        long diffTimeStamp = bridgeConstants.getMinSecondsBetweenCallsToReceiveHeader();
+
+        long currentTimeStamp = rskExecutionBlock.getTimestamp(); //in seconds
+        Optional<Long> optionalLastTimeStamp = provider.getReceiveHeadersLastTimestamp();
+        if (optionalLastTimeStamp.isPresent() && (currentTimeStamp - optionalLastTimeStamp.get() < diffTimeStamp)) {
+            logger.warn("Receive header last TimeStamp less than {} milliseconds", diffTimeStamp);
+            return RECEIVE_HEADER_CALLED_TOO_SOON;
+        }
+
+        //Depth
+        StoredBlock previousBlock = btcBlockStore.get(header.getPrevBlockHash());
+        if (previousBlock == null) {
+            return RECEIVE_HEADER_CANT_FOUND_PREVIOUS_BLOCK;
+        }
+
+        // height of best chain - height of current header block greater than maximum depth accepted
+        if ((getBtcBlockchainBestChainHeight() - (previousBlock.getHeight() + 1)) > bridgeConstants.getMaxDepthBlockchainAccepted()) {
+            return RECEIVE_HEADER_BLOCK_TOO_OLD;
+        }
+
+        try {
+            btcBlockChain.add(header);
```

Comment:
There's a bridge performance improvement that needs to cross check its functionality with this.

If this code advance the blockchain to another branch, are the intermediate blocks marked correctly? I asked something about this to marcos